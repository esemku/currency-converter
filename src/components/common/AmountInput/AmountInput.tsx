import React from 'react';
import CurrencyInput from 'react-currency-input-field';
import useStyles from './styles';

interface IProps {
  inputName: string;
  label: string;
  onChange: any;
}

const AmountInput: React.FC<IProps> = ({ inputName, label, onChange }) => {
  const styles = useStyles();

  return (
    <div className={styles.root}>
      <label htmlFor={inputName} className={styles.label}>
        {label}
      </label>
      <CurrencyInput
        id={inputName}
        name={inputName}
        placeholder="e.g.  1,000.00"
        decimalScale={2}
        onValueChange={onChange}
        decimalSeparator="."
        groupSeparator=","
        className={styles.input}
      />
    </div>
  );
};

export default AmountInput;
