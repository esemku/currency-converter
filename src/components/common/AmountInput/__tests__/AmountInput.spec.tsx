import { render } from '@testing-library/react';
import React from 'react';
import AmountInput from '../AmountInput';

describe('<AmountInput />', () => {
  it('renders correctly', () => {
    const { container } = render(
      <AmountInput inputName="inputName" label="label" onChange={jest.fn()} />,
    );

    expect(container).toMatchSnapshot();
  });
});
