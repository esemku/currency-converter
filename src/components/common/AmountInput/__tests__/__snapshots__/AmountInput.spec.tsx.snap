// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AmountInput /> renders correctly 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <label
      class="makeStyles-label-3"
      for="inputName"
    >
      label
    </label>
    <input
      class="makeStyles-input-2"
      data-testid="amountInput"
      id="inputName"
      inputmode="decimal"
      name="inputName"
      placeholder="e.g.  1,000.00"
      type="text"
      value=""
    />
  </div>
</div>
`;
